topology/consistency_level.go

```diff
@@ -307,3 +307,68 @@ const (
 	majority         = "majority"
 	unstrictMajority = "unstrict_majority"
 )
+
+// WriteConsistencyAchieved returns a bool indicating whether or not we've received enough
+// successful acks to consider a write successful based on the specified consistency level.
+func WriteConsistencyAchieved(
+	level ConsistencyLevel,
+	majority, numPeers, numSuccess int,
+) bool {
+	switch level {
+	case ConsistencyLevelAll:
+		if numSuccess == numPeers { // Meets all
+			return true
+		}
+	case ConsistencyLevelMajority:
+		if numSuccess >= majority { // Meets majority
+			return true
+		}
+	case ConsistencyLevelOne:
+		if numSuccess > 0 { // Meets one
+			return true
+		}
+	default:
```

Comment:
unrelated to your change, but mind dropping the default and changing the return false to a panic (same as two functions below)